olp-cpp-sdk-core/src/client/OlpClient.cpp

```diff
@@ -172,13 +199,16 @@ HttpResponse SendRequest(const http::NetworkRequest& request,
   if (!condition.Wait(std::chrono::seconds(retry_settings.timeout))) {
     OLP_SDK_LOG_INFO_F(kLogTag, "Timeout");
     context.CancelOperation();
+    while (interest_flag->load() == State::Processing) {
```

Comment:
What if the state is Waiting or Done? This will be an endless loop.
Also, it seems to me that the context.CancelOperation() called one line before the check is setting the flag to State::Done actually, right? Except of course it is in the state Processing.